app/models/subject.rb

```diff
@@ -1,2 +1,11 @@
 class Subject < ApplicationRecord
+  enum status: {pending: 0, started: 1, finished: 2}
+  VALID_NAME_REGEX = /\A[a-zA-Z0-9 ]+\z/i.freeze
+
+  validates :name, presence: true,
+                   allow_blank: false,
+                   format: { with: VALID_NAME_REGEX }
+  validates :status, presence: true,
+                     numericality: true,
+                     inclusion: { in: statuses }
```

Comment:
Nếu xài inclusion để bắt chặt chẽ hơn thì như này mới đúng e nhé
`inclusion: { in: statuses.keys }`